app/js/datepicker.jsx

```diff
@@ -66,6 +66,27 @@ class DatePicker extends React.Component {
 
     componentDidMount() {}
 
+    componentWillReceiveProps(newProps) {
+        if (newProps.defaultEndDate !== this.props.defaultEndDate ||
+            newProps.defaultDate !== this.props.defaultDate) {
+            let endDate = newProps.defaultEndDate;
+            let startDate = newProps.defaultDate;
+
+            if(!startDate) {
+                startDate = moment();
+            }
+
+            if (!endDate) {
+                endDate = startDate.add(1, "months");
```

Comment:
so, this doesn't introduce a bug, but it has the same bug as we have in the contructor. We need to copy startdate, otherwise we're modifying it underneath. This sets startdate ahead a month of where we actually want it. I've updated this and added a demo to index.